src/Microsoft.DotNet.Wpf/src/PresentationFramework/Microsoft/Win32/FileDialog.cs

```diff
@@ -329,10 +324,11 @@ public string Filter
                         // This implicitly requires there to be at least one vertical bar in
                         // the filter string - or else formats.Length will be 1, resulting in an
                         // ArgumentException.
+                        int formatsCount = 0;
+                        foreach (Range range in updatedFilter.AsSpan().Split('|'))
+                            formatsCount++;
```

Comment:
Since we only use the SpanSplitEnumerator to derive a count, it may be a little faster to work with [`updatedFilter.AsSpan().Count('|')`](https://learn.microsoft.com/en-us/dotnet/api/system.memoryextensions.count?view=net-8.0#system-memoryextensions-count-1(system-readonlyspan((-0))-0)) for most filter strings (anything longer than 8 characters will use a vectorised implementation.)

The `formatsCount` logic would then naturally change slightly to `formatsCount % 2 == 0`, since we're counting separators rather than segments.

Reply:
It's only gonna be five times faster than using the struct enumerator, that's nothing worth talking about.

On a serious note, thank you for this review, I totally forgot we have now vectorized `Count` in `MemoryExtensions` as I didn't have a chance to look at it and it just brings LINQ flashbacks. Will address.